app/views/sessions/new.html.erb

```diff
@@ -1,7 +1,17 @@
 <h1>Login</h1>
 
+<%= flash[:notice] %>
 
-<%= form_tag '/login' do %>
+<% if !!flash[:alert]%>
+  <% flash.each do |k, msg| %>
```

Comment:
Not sure about how this works, could you explain ?

Reply:
not familiar with flash notices

Reply:
Which bit? If a certain error or alert worthy scenario occurs, in the controller we set a`flash[:alert]` variable, then this here is just a for each loop to display them to the user